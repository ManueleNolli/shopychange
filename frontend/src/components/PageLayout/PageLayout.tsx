import React from 'react'

import { Box } from '@chakra-ui/react'

import Header from './Header/Header'
import Footer from './Footer/Footer'
import { ACCOUNT_NAV_ITEMS, NAV_ITEMS } from '../../data/data'

type TemplateProps = {
  children: React.ReactNode
  header?: boolean
  footer?: boolean
}

export default function PageLayout({
  children,
  header = true,
  footer = true,
}: TemplateProps) {
  return (
    <Box minH="100vh" display="flex" flexFlow="column nowrap">
      {header && (
        <Header navItems={NAV_ITEMS} accountItems={ACCOUNT_NAV_ITEMS} />
      )}
      <Box flexGrow={1}>{children}</Box>
      {footer && <Footer />}
    </Box>
  )
}
