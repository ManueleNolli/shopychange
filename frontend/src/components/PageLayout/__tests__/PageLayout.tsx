import React from 'react'
import { render } from '@testing-library/react'
import PageLayout from '../PageLayout'

jest.mock('../Header/Header')

jest.mock('../Footer/Footer')

jest.mock('@wagmi/core', () => ({
  disconnect: jest.fn(),
}))

describe('PageLayout', () => {
  test('renders correctly', () => {
    const tree = render(
      <PageLayout>
        <div data-testid={'children'}>Children</div>
      </PageLayout>
    )
    expect(tree.container).toMatchSnapshot()
  })
  test('renders correctly without header', () => {
    const tree = render(
      <PageLayout header={false}>
        <div data-testid={'children'}>Children</div>
      </PageLayout>
    )
    expect(tree.container).toMatchSnapshot()
  })
  test('renders correctly without footer', () => {
    const tree = render(
      <PageLayout footer={false}>
        <div data-testid={'children'}>Children</div>
      </PageLayout>
    )
    expect(tree.container).toMatchSnapshot()
  })
  test('renders correctly without footer and header', () => {
    const tree = render(
      <PageLayout footer={false} header={false}>
        <div data-testid={'children'}>Children</div>
      </PageLayout>
    )
    expect(tree.container).toMatchSnapshot()
  })
})
