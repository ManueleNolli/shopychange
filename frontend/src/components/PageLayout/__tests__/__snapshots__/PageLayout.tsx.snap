// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout renders correctly 1`] = `
<div>
  <div
    class="css-1m0xqts"
  >
    <div
      data-testid="HeaderContent"
    />
    <div
      class="css-i9gxme"
    >
      <div
        data-testid="children"
      >
        Children
      </div>
    </div>
    <div
      data-testid="Footer"
    />
  </div>
</div>
`;

exports[`PageLayout renders correctly without footer 1`] = `
<div>
  <div
    class="css-1m0xqts"
  >
    <div
      data-testid="HeaderContent"
    />
    <div
      class="css-i9gxme"
    >
      <div
        data-testid="children"
      >
        Children
      </div>
    </div>
  </div>
</div>
`;

exports[`PageLayout renders correctly without footer and header 1`] = `
<div>
  <div
    class="css-1m0xqts"
  >
    <div
      class="css-i9gxme"
    >
      <div
        data-testid="children"
      >
        Children
      </div>
    </div>
  </div>
</div>
`;

exports[`PageLayout renders correctly without header 1`] = `
<div>
  <div
    class="css-1m0xqts"
  >
    <div
      class="css-i9gxme"
    >
      <div
        data-testid="children"
      >
        Children
      </div>
    </div>
    <div
      data-testid="Footer"
    />
  </div>
</div>
`;
